core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -91,17 +103,25 @@ private[spark] class HeartbeatReceiver(sc: SparkContext, scheduler: TaskSchedule
         scheduler.executorLost(executorId, SlaveLost("Executor heartbeat " +
           s"timed out after ${now - lastSeenMs} ms"))
         if (sc.supportDynamicAllocation) {
-          sc.killExecutor(executorId)
+          // Asynchronously kill the executor to avoid blocking the current thread
```

Comment:
was it not doing async before? if it wasn't, why do we need to change it here? If it was, did the RPC contract changed it so that it is no longer async?